### ISSUE TITLE
Incorrect focus behavior of bookmark and link toolbars

### ISSUE THREAD

**Issue by Comandeer** [ckeditor/ckeditor5#18336]
## 📝 Provide detailed reproduction steps (if any)

1. Go to https://ckeditor.com/docs/ckeditor5/latest/features/bookmarks.html
2. Select any bookmark.
3. Click the bookmark toolbar button.

### ✔️ Expected result

The focus should be moved to the bookmark toolbar.

### ❌ Actual result

The focus stays in the same place.

![Image](https://github.com/user-attachments/assets/af28ebe2-1c31-4dca-81ed-9b8888abe166)

It's extremely confusing for the users of screen readers as it seems like pressing the button does nothing.

The same happens with links and link toolbar (also when using the <kbd>Cmd</kbd>+<kbd>K</kbd> shortcut).

### ❓ Possible solution

The focus should be moved to the bookmark/link toolbar when it's shown.

## 📃 Other details

* Browser: Chrome 134.0.6998.166 
* OS: macOS 15.4
* First affected CKEditor version: 44.3.0
* Installed CKEditor plugins: `bookmark`, `link`

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.
